job-server/src/spark.jobserver/JobServer.scala

```diff
@@ -66,6 +75,44 @@ object JobServer {
 
   }
 
+  private def storeInitialJars(config: Config, jarManager: ActorRef): Unit = {
+    if(config.hasPath("spark.jobserver.job-jar-uris")) {
+      val initialJarsConfig = config.getConfig("spark.jobserver.job-jar-uris").root
+
+      logger.info("Adding initial job jars: {}", initialJarsConfig.render())
+
+      val initialJars =
+        initialJarsConfig
+          .asScala
+          .map { case (key, value) => (key, value.unwrapped.toString) }
+          .toMap
+
+      // Ensure that the jars exist
+      for(jarPath <- initialJars.values) {
+        val f = new java.io.File(jarPath)
+        if(!f.exists) {
+          val msg =
+            if(f.isAbsolute) {
+              s"Inital Jar File $jarPath does not exist"
+            } else {
+              s"Inital Jar File $jarPath (${f.getAbsolutePath}) does not exist"
+            }
+
+          throw new java.io.IOException(msg)
```

Comment:
Does this means that it is all or none?

For exampl, say I have four apps and corresponding paths and somehow one of the path is failing. Does this mean none them will be uploaded?